diff --git a/src/go/Makefile.am b/src/go/Makefile.am
index 2dd9e84e4ea..74bc3fbb91e 100644
--- a/src/go/Makefile.am
+++ b/src/go/Makefile.am
@@ -49,7 +49,7 @@ endif
 all: build
 
 zabbix.com/cmd/zabbix_agent2:
-	CGO_CFLAGS="${CGO_CFLAGS}" CGO_LDFLAGS="${CGO_LDFLAGS}" go build -tags="${AGENT_GOTAGS}" -ldflags="${AGENT_GOLDFLAGS}" -o bin zabbix.com/cmd/zabbix_agent2
+	CGO_CFLAGS="${CGO_CFLAGS} -D_LARGEFILE64_SOURCE" CGO_LDFLAGS="${CGO_LDFLAGS}" go build -tags="${AGENT_GOTAGS}" -ldflags="${AGENT_GOLDFLAGS}" -o bin zabbix.com/cmd/zabbix_agent2
 
 zabbix.com/cmd/zabbix_web_service:
 	go build -ldflags="${WEBSERVICE_GOLDFLAGS}" -o bin zabbix.com/cmd/zabbix_web_service
