diff --git a/src/go/cmd/zabbix_web_service/pdf_report_creator.go b/src/go/cmd/zabbix_web_service/pdf_report_creator.go
index afab41204d1..f1a9b859f34 100644
--- a/src/go/cmd/zabbix_web_service/pdf_report_creator.go
+++ b/src/go/cmd/zabbix_web_service/pdf_report_creator.go
@@ -100,7 +100,11 @@ func (h *handler) report(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	opts := chromedp.DefaultExecAllocatorOptions[:]
+	opts := append(chromedp.DefaultExecAllocatorOptions[:],
+                chromedp.DisableGPU,
+                chromedp.NoSandbox,
+                chromedp.Flag("disable-setuid-sandbox", true),
+        )
 
 	if options.IgnoreURLCertErrors == 1 {
 		opts = append(opts, chromedp.Flag("ignore-certificate-errors", "1"))
